### ISSUE TITLE
low performance for kernel with padded dim and RNG Ops

### ISSUE THREAD

**Issue by liqiangxl** [NVIDIA/Fuser#1084]
In this case, bdimx is padded from 106 to 128 and the prediate in else branch for RNG Ops is removed. This lead to performance regression compared with unpadded version as discussed in https://github.com/NVIDIA/Fuser/pull/988. Performance data for different tests:

Test | latency, ms
-- | --
padded (main branch) | 0.74
unpadded | 0.56
padded with predicate | 0.58
unpadded with predicate | 0.56

To solve this issue, we can:
(1) don't eliminate predicate for RNG Ops
(2) use unpadded kernel.

Script to reproduce:

```
import math
import torch
from nvfuser import FusionDefinition, DataType
from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype


def nvfuser_scale_mask_softmax_dropout(fd: FusionDefinition, shape, input_dtype, prob=0.20) -> None:
    n_head = shape[1]
    embed_dim_per_head = 64
    batch_size = shape[0]
    seq_len = shape[-1]
    r_sqrt_nhead = 1.0 / math.sqrt(embed_dim_per_head)

    # batch size, head count, sequence length, sequence length
    T0 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=input_dtype, is_cpu=False)
    T1 = fd.define_tensor(shape=[-1, -1], contiguity=[True, True], dtype=input_dtype, is_cpu=False)
    if input_dtype == DataType.Half:
        T0 = fd.ops.cast(T0, dtype=DataType.Float)
        T1 = fd.ops.cast(T1, dtype=DataType.Float)
    T1 = fd.ops.broadcast_in_dim(T1, shape=shape, broadcast_dims=[2, 3])
    S2 = fd.define_scalar(r_sqrt_nhead, dtype=DataType.Double)
    T3 = fd.ops.mul(T0, S2)
    T4 = fd.ops.slice(T1, start_indices=[0, 0, 0, 0], end_indices=[1, 1, seq_len, seq_len], strides=[1, 1, 1, 1])
    S5 = fd.define_scalar(0.00000, dtype=DataType.Double)
    T6 = fd.ops.eq(S5, T4)
    T7 = fd.ops.broadcast_in_dim(T6, shape=[batch_size, n_head, seq_len, seq_len], broadcast_dims=[0, 1, 2, 3])
    S8 = fd.define_scalar(float("-inf"), dtype=DataType.Double)
    T9 = fd.ops.where(T7, S8, T3)
    T13 = fd.ops.max(T9, axes=[3], keepdim=True, dtype=DataType.Null)
    T16 = fd.ops.sub(T9, T13)
    T17 = fd.ops.exp(T16)
    T21 = fd.ops.sum(T17, axes=[3], keepdim=True, dtype=DataType.Null)
    r_T21 = fd.ops.reciprocal(T21)
    T24 = fd.ops.mul(T17, r_T21)
    S25 = fd.define_scalar(batch_size, dtype=DataType.Int)
    S26 = fd.define_scalar(n_head, dtype=DataType.Int)
    S27 = fd.define_scalar(seq_len, dtype=DataType.Int)
    S28 = fd.define_scalar(seq_len, dtype=DataType.Int)
    S29 = fd.define_scalar(0.00000, dtype=DataType.Double)
    S30 = fd.define_scalar(1.00000, dtype=DataType.Double)
    T31 = fd.ops.uniform(S29, S30, shape=[S25, S26, S27, S28], dtype=DataType.Float)
    S32 = fd.define_scalar(1.0 - prob, dtype=DataType.Double)
    T33 = fd.ops.lt(T31, S32)
    T34 = fd.ops.cast(T33, dtype=DataType.Float)
    T35 = fd.ops.mul(T24, T34)
    S36 = fd.define_scalar(1.0 / (1.0 - prob), dtype=DataType.Double)
    T37 = fd.ops.mul(T35, S36)

    if input_dtype == DataType.Half:
        T37 = fd.ops.cast(T37, dtype=DataType.Half)

    fd.add_output(T37)


if __name__ == "__main__":
    torch_dtype = torch.half
    batch = 2
    seq_len = 4224
    n_head = 2
    inputs = [
        torch.randn(batch, n_head, seq_len, seq_len, dtype=torch_dtype, device="cuda"),
        torch.randn(seq_len, seq_len, dtype=torch_dtype, device="cuda")
    ]

    with FusionDefinition() as fd:
        nvfuser_scale_mask_softmax_dropout(fd, inputs[0].shape, torch_dtype_to_nvfuser_dtype(torch_dtype))

    nvfuser_output = fd.execute(inputs)

```
